src/grid_read.rs

```diff
@@ -0,0 +1,46 @@
+use std::fs;
+use std::path::PathBuf;
+
+use crate::Grid;
+
+pub fn file_read(filepath: PathBuf) -> String {
+    let content = fs::read_to_string(filepath).expect("Problème lors de la lecture de la grille");
+    content
+}
+
+pub fn size(content: &String) -> &str {
+    let bytes = &content.as_bytes();
+
+    for (i, &element) in bytes.iter().enumerate() {
+        if element == b'\n' {
+            return &content[0..i];
+        }
+    }
+    panic!("Fichier incorrect : size");
+}
+
+pub fn fill_grid_from_file(grid: &mut Grid, content: &String) {
+    let bytes = &content.as_bytes();
+    let mut x: usize = 0;
+    let mut y: usize = 0;
+    let mut s: bool = false;
+
+    for (_i, &element) in bytes.iter().enumerate() {
```

Comment:
Pourquoi `enumerate()` tu si c'est pour ne rien faire de l'index `_i` ?
Autant juste écrire:
```suggestion
    for &element in bytes.iter() {
```

Reply:
C'est vrai que c'est idiot mais ça vient pas du tout du fait que j'ai piqué ça sur le livre de Rust sûrement pas